pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -384,15 +421,23 @@ func (r *repository) Put(manifest *schema1.SignedManifest) error {
 			return statusErr
 		}
 
-		if _, err := client.ImageStreams(r.namespace).Create(&stream); err != nil {
+		is, err := client.ImageStreams(r.namespace).Create(&stream)
+		if err != nil {
 			context.GetLogger(r.ctx).Errorf("Error auto provisioning image stream: %s", err)
 			return statusErr
 		}
 
 		// try to create the ISM again
-		if err := r.registryClient.ImageStreamMappings(r.namespace).Create(&ism); err != nil {
-			context.GetLogger(r.ctx).Errorf("Error creating image stream mapping: %s", err)
+		if err := r.registryOSClient.ImageStreamMappings(r.namespace).Create(&ism); err != nil {
+			statusErr, ok := err.(*kerrors.StatusError)
+			if ok && kerrors.IsForbidden(statusErr) {
+				context.GetLogger(r.ctx).Errorf("Denied creating ImageStreamMapping: %v", statusErr)
+				reportQuotaExceededEvent(r.ctx, r, is, "FailedUpdate", statusErr.Error())
```

Comment:
This isn't necessarily about quota failing, though it is the mostly cause.  Is there any better introspection you can do?